### PR TITLE
chore(pre-commit): Cleanup various warnings from pre-commit hooks

### PR DESCRIPTION
**Pull request by cfxegbert**

### CODE REVIEW

**Comment by cfxegbert** [comment]
Since my other pull request I fixed one pre-commit issue I decided to clean up all the pre-commit issues.

I have not looked at upgrading to ruff >= 0.3.x.  Should I look at upgrading it?

**Comment by codejedi365** [comment]
Sure, if you want to wrap in #856 that would be good.

**Comment by cfxegbert** [comment]
Ruff upgraded to 0.3.2.  Should I upgrade the other pre-commit hooks?

File: tests/unit/semantic_release/version/test_algorithm.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def test_bfs_for_latest_version_in_history():
     v1_commit = Commit(repo, binsha=b"0" * 20)
 
     class TagReferenceOverride(TagReference):
-        commit = v1_commit  # type: ignore - mocking the commit property
+        commit = v1_commit  # mocking the commit property
```
Comment (codejedi365) [inline tests/unit/semantic_release/version/test_algorithm.py:32]: Is mypy ok with this?

**Review by codejedi365 — COMMENTED** [review]
Got a review in already, if you want to make the update of the other pre commit go ahead. Only concern is that we have quite a few PRs waiting for merge. Likely this will be delayed until after all of them as this will likely cause more conflicts from basic formatting.

File: tests/unit/semantic_release/version/test_algorithm.py (hunk 1 above)
Comment (cfxegbert) [reply tests/unit/semantic_release/version/test_algorithm.py:32]: mypy is fine with this even with the `--strict` option

Comment (codejedi365) [reply tests/unit/semantic_release/version/test_algorithm.py:32]: Interesting... I guess it got updated. I thought it was weird the first time I had to add it

**Comment by cfxegbert** [comment]
> Likely this will be delayed until after all of them as this will likely cause more conflicts from basic formatting.

I'll revert the style change commit and modify the ruff configuration to not change the style.  We can then figure out a window when we want to upgrade the style.

**Comment by codejedi365** [comment]
Rock on 🤘 thanks!

**Comment by cfxegbert** [comment]
Fixed commit messages not to begin with a capital leter

**Review by codejedi365 — APPROVED** [review]
Great job. 

Ready for merge, @relekang

**Comment by cfxegbert** [comment]
Fixed log message to be under 100 characters

**Pull request merged**